src/codemodder/code_directory.py

```diff
@@ -36,13 +36,26 @@ def file_line_patterns(file_path: str | Path, patterns: Sequence[str]):
     ]
 
 
-def filter_files(names: Sequence[str], patterns: Sequence[str], exclude: bool = False):
+def filter_files(
+    names: Sequence[str],
+    parent_path: str,
+    patterns: Sequence[str],
+    exclude: bool = False,
+):
     patterns = (
         [x.split(":")[0] for x in (patterns or [])]
         if not exclude
         # An excluded line should not cause the entire file to be excluded
         else [x for x in (patterns or []) if ":" not in x]
     )
+
+    # TODO: handle case when parent path is "."
+    patterns = [
+        str(Path(parent_path) / Path(pat))
+        if not pat.startswith("*")
+        else parent_path + pat
+        for pat in patterns
+    ]
```

Comment:
I'd like to combine and avoid doing multiple list comp operations but it was honestly easier to iterate and easier to understand separating it. We can refactor later on with a generator or something.

Also, I don't feel super confident with this `if not pat.startswith("*")` logic, but tests show it works right now. happy to revisit

Reply:
Suggestion:
I think you can use a combination of `glob` and `pathlib` to solve issues with globs and relative paths
```
full_path = (Path(parent_path) / Path(pat)).resolve()
files = glob.glob(str(full_path))
```
There are two caveats: (1) resolve will also make it absolute, we may need to handle the current directory carefully, (2) glob will return actual files in the filesystem which is slower than `fnmatch`